src/Psl/DateTime/DateTimeConvenienceMethodsTrait.php

```diff
@@ -438,25 +442,25 @@ public function minusMonths(int $months): static
             return $this->plusMonths(-$months);
         }
 
-        $current_year = $this->getYear();
-        $current_month = $this->getMonthEnum();
-        $days_to_subtract = 0;
-        for ($i = 0; $i < $months; $i++) {
-            // When subtracting, we need to move the current month back before the calculation
-            $total_months = $current_month->value - $i;
-            while ($total_months <= 0) {
-                $total_months += MONTHS_PER_YEAR; // Adjust month to be within 1-12
-                $current_year--; // Adjust year when wrapping
-            }
-
-            $target_month = ($total_months % MONTHS_PER_YEAR) ?: MONTHS_PER_YEAR;
-            $target_year = $current_year + Math\div($total_months - 1, MONTHS_PER_YEAR);
-
-            // Subtract days of the month we are moving into
-            $days_to_subtract += Month::from($target_month)->getDaysForYear($target_year);
+        $minus_years = Math\div($months, 12);
+        $months_left = $months - ($minus_years * 12);
```

Comment:
todo: `12` should be replaced with `MONTHS_PER_YEAR` ( just in case humanity decided to change the calender, so we have less work todo 😛 )